llvm/lib/Support/TimeProfiler.cpp

```diff
@@ -114,16 +114,16 @@ struct llvm::TimeTraceProfiler {
 
     // Check that end times monotonically increase.
     assert((Entries.empty() ||
-            (E.getFlameGraphStartUs(StartTime) + E.getFlameGraphDurUs() >=
-             Entries.back().getFlameGraphStartUs(StartTime) +
-                 Entries.back().getFlameGraphDurUs())) &&
+            (E.getFlameGraphStartNs(StartTime) + E.getFlameGraphDurNs() >=
+             Entries.back().getFlameGraphStartNs(StartTime) +
+                 Entries.back().getFlameGraphDurNs())) &&
            "TimeProfiler scope ended earlier than previous scope");
 
     // Calculate duration at full precision for overall counts.
     DurationType Duration = E.End - E.Start;
 
     // Only include sections longer or equal to TimeTraceGranularity msec.
-    if (duration_cast<microseconds>(Duration).count() >= TimeTraceGranularity)
+    if (duration_cast<nanoseconds>(Duration).count() >= TimeTraceGranularity)
```

Comment:
TimeTraceGranularity is in microseconds wherever timeTraceProfilerInitialize gets called.  Is there a plan to update the interface for compatibility (i.e. adjust microseconds to nanoseconds internally)?

Reply:
I didn't think about it, but I can do it. We need a easy way for users, to change it. Do you have an idea of how to do it?

Reply:
Hi Mark, I've been looking at this, but now I'm not sure what you mean. After looking at the tracing format ( Perfetto/chronium ) I see that it requires that the events duration have to be passed in microseconds. I made this change because I thought it was possible to provide trace duration in nanoseconds. I can obtain basically the same behavior on on the libomptarget trace by changing the TimeTraceGranularity at the construction of the trace, to 0. Given this, at this point I think that this patch is not necessary. But maybe I'm missing something, what do you mean with "update the interface for compatibility".

Reply:
This silently changes the meaning of TimeTraceGranularity from microseconds to nanoseconds, and there are users of the code besides OpenMP, and it changes the meaning for them, too.
I usually add an environment variable (LIBOMPTARGET_PROFILE_GRANULARITY) to change the granularity, and set it to zero for profiling.  I thought there was a review adding such a variable, but I can't find it now.